src/collision.py

```diff
@@ -0,0 +1,51 @@
+from __future__ import annotations
+
+import arcade
+
+from bullet import Projectile
+from linked_sprite import LinkedSprite
+from player import Player
+
+
+def projectile_hits_wall(projectile_spritelist, wall_spritelist):
+    for projectile in projectile_spritelist:
+        projectile: LinkedSprite[Projectile]
+        walls_touching_projectile = arcade.check_for_collision_with_list(
+            projectile, wall_spritelist
+        )
+
+        if len(walls_touching_projectile) > 0:
+            projectile.owner.on_collision_with_wall(
+                projectile_spritelist, walls_touching_projectile
+            )
+
+
+def projectile_hits_player(projectile_spritelist, player_spritelist):
```

Comment:
You can pass this function `sprite_lists` as the argument, and you can call `sprite_lists.projectiles` and `sprite_lists.walls`